src/util.ts

```diff
@@ -91,7 +91,7 @@ export function createOutputRetainingCompilerHost(
 }
 
 export function extractInlineSourceMap(source: string): string {
-  const regex = new RegExp('//# sourceMappingURL=data:application/json;base64,(.*)');
+  const regex = new RegExp('^//# sourceMappingURL=data:application/json;base64,(.*)$', 'm');
   const result = regex.exec(source)!;
```

Comment:
Tsc should always produce the same source map comment, so since we're protected by a check that the proper flag was passed we should be good.